core/src/main/java/com/linecorp/armeria/server/docs/EndpointInfo.java

```diff
@@ -30,43 +30,39 @@
 import com.google.common.base.MoreObjects;
 import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableSortedSet;
-import com.google.common.collect.Streams;
 
 import com.linecorp.armeria.common.MediaType;
-import com.linecorp.armeria.common.SerializationFormat;
 import com.linecorp.armeria.server.Service;
 
 /**
  * Metadata about the endpoints exposed by a {@link Service}.
  */
+@JsonInclude(Include.NON_NULL)
 public final class EndpointInfo {
 
     private final String hostnamePattern;
     private final String path;
+
+    @Nullable
+    private final String regexPathPrefix;
     @Nullable
     private final String fragment;
+    @Nullable
     private final MediaType defaultMimeType;
     private final Set<MediaType> availableMimeTypes;
 
     /**
      * Creates a new instance.
      */
-    public EndpointInfo(String hostnamePattern, String path, @Nullable String fragment,
-                        SerializationFormat defaultFormat, Iterable<SerializationFormat> availableFormats) {
-        this(hostnamePattern, path, fragment, defaultFormat.mediaType(),
-             Streams.stream(availableFormats).map(SerializationFormat::mediaType)::iterator);
-    }
-
-    /**
-     * Creates a new instance.
-     */
-    public EndpointInfo(String hostnamePattern, String path, @Nullable String fragment,
-                        MediaType defaultMimeType, Iterable<MediaType> availableMimeTypes) {
+    public EndpointInfo(String hostnamePattern, String path, @Nullable String regexPathPrefix,
+                        @Nullable String fragment, @Nullable MediaType defaultMimeType,
+                        Iterable<MediaType> availableMimeTypes) {
 
         this.hostnamePattern = requireNonNull(hostnamePattern, "hostnamePattern");
         this.path = requireNonNull(path, "path");
+        this.regexPathPrefix = Strings.emptyToNull(regexPathPrefix);
```

Comment:
`path` is originally designed to provide the 'exact' path only. Perhaps we should generalize this a little bit, like:

- renaming `path` to `pathMapping`
- adding path mapping type? e.g. exact, prefix, glob, regex, ...

(or just accepting `PathMapping`? Hmm..)